### PR TITLE
Adopt the same output strategy as for the GCS and SA modules

### PR DESCRIPTION
**Pull request by ludoo**
This PR changes outputs so that they follow the same strategy used in the `cloud-storage` and `service-accounts` folder, namely:

- add bare resource outputs, now possible with 0.12
- add a full set of single-use outputs
- add `_list` outputs exposing unordered lists
- consistent naming

This is of course a backwards incompatible change, so it makes sense to release it together with [#8].

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -14,13 +14,42 @@
...
 
-output "names_and_display_names" {
-  description = "Map of name => folder resource name."
-  value       = zipmap(var.names, google_folder.folders.*.display_name)
+output "folder" {
+  description = "Folder resource (for single use)."
+  value       = google_folder.folders[0]
 }
 
-output "names_and_ids" {
```
Comment (morgante) [inline outputs.tf]: I'm not sure we want to make this a breaking change by renaming the outputs?

If we do please include a short migration guide.

Comment (ludoo) [reply outputs.tf]: Added migration guide for changed/new outputs, and parent variable change.

**Comment by ludoo** [comment]
Added migration instructions, ready for a second review pass.

**Comment by morgante** [comment]
LGTM, merge and release at will.

**Pull request merged**